src/main.rs

```diff
@@ -97,11 +98,36 @@ impl LanguageServer for Backend {
         ])))
     }
 
-    async fn hover(&self, _: HoverParams) -> Result<Option<Hover>> {
-        Ok(Some(Hover {
-            contents: HoverContents::Scalar(MarkedString::String("You're hovering!".to_string())),
-            range: None,
-        }))
+    async fn hover(&self, params: HoverParams) -> Result<Option<Hover>> {
+        let world = self.world.read().await;
+        let world = world.as_ref().unwrap();
+        let source = world.main();
+
+        let cursor = get_cursor_for_position(params.text_document_position_params.position, source);
+
+        if let Some(cursor) = cursor {
```

Comment:
I'm just dreaming of chain if let tbh

Reply:
I could use if_chain! if you'd like !

Reply:
@astrale-sharp can you implement something like @jakobhellermann's idea here? It will keep the code from getting too deeply nested.